lib/net.c

```diff
@@ -1119,6 +1119,63 @@ start_iface(struct gatekeeper_if *iface)
 	return ret;
 }
 
+static int
+init_net_stage1(void *arg)
+{
+	struct net_config *net_conf = arg;
+	uint32_t i;
+
+	if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
+		net_conf->gatekeeper_pktmbuf_pool =
+			rte_calloc("mbuf_pool", net_conf->numa_nodes,
+				sizeof(struct rte_mempool *), 0);
+		if (net_conf->gatekeeper_pktmbuf_pool == NULL) {
+			RTE_LOG(ERR, MALLOC, "%s: Out of memory\n", __func__);
+			return -1;
+		}
+	}
+
+	/* Initialize pktmbuf pool on each used NUMA node. */
+	for (i = 0; i < net_conf->numa_nodes; i++) {
+		char pool_name[64];
+		int ret;
+
+		if (!net_conf->numa_used[i] ||
+				net_conf->gatekeeper_pktmbuf_pool[i] != NULL)
```

Comment:
As explained in the previous comment, we will now initialize the memory pools in stage 1 instead of during a call to `gatekeeper_init_network()`, so we can simplify the test above to `!net_conf->numa_used[i]`.

Reply:
Discard this comment because we are not cleaning up if a failure occurs.